domain/src/main/java/com/teamagam/gimelgimel/domain/layers/ProcessNewVectorLayerInteractor.java

```diff
@@ -62,39 +62,62 @@
     protected Iterable<SubscriptionRequest> buildSubscriptionRequests(
             DataSubscriptionRequest.SubscriptionRequestFactory factory) {
         DataSubscriptionRequest dataSubscriptionRequest = factory.create(
-                buildProcessingObservable());
+                buildObservable());
         return Collections.singletonList(dataSubscriptionRequest);
     }
 
-    private Observable<URI> buildProcessingObservable() {
-        return Observable.just(null)
-                .flatMap(x -> fetchCachedURI())
-                .doOnNext(uri -> sLogger.i("VectorLayer cached uri:" + uri.toString()))
+    private Observable<URI> buildObservable() {
+        return Observable.just(mVectorLayer)
+                .flatMap(this::processIfNeeded);
+    }
+
+    private Observable<URI> processIfNeeded(VectorLayer vl) {
+        if (isOutdatedVectorLayer(vl)) {
+            sLogger.d("Not processing following vector layer because it's outdated: " + vl);
+            return Observable.empty();
+        }
+        sLogger.d("Processing vector layer " + vl);
+        return buildProcessObservable(vl);
+    }
+
+    private boolean isOutdatedVectorLayer(VectorLayer vl) {
+        return mVectorLayerRepository.contains(vl.getId()) &&
```

Comment:
Is it "clean" to depend on short circuit evaluation? I ask because I don't know...
I needed it once myself and decided not to use it.
Anyway, if it is OK, there is some place in the code that needs to be refactored but the problem is that I don't remember where...